### DISCUSSION TITLE
Dynamic index and query with exact() and startsWith()

### DISCUSSION THREAD

**Discussion by harrison314** [ravendb/ravendb#14709] [Q&A] [answered]
Hello,
is it possible to create a dynamic index so that _exact()_ and _startsWith()_ can be used in the query?

I tried all combinations of fields in `CreateFieldOptions`, but `exact` or `startsWith` didn't work (in RavenDb v. 5.3 and 5.4).

Unit test for replication: https://gist.github.com/harrison314/41b1dccc48520636b5b212c24fd65f72

Would a custom analyser help?

**Comment by garayx** [comment, marked as answer]
Hi 

RavenDB uses `LowerCaseKeywordAnalyzer` by [default](https://ravendb.net/docs/article-page/5.3/csharp/indexes/using-analyzers#ravendb), so in order to match the `FieldIndexing.Exact` option in `CreateFieldOptions` of Index, you will have to wrap the `startsWith()` with `Exact()` method:

`from index 'LogEntityIndex' where exact(startsWith(Application, 'Area52'))`

**Comment by harrison314** [comment]
Thanks for the answer.
Well, I needed `stratWith()` case-insensitive.

I managed to achieve this way:

```
public class LogEntityIndex : AbstractIndexCreationTask<LogEntity>
{
    public LogEntityIndex()
    {
        this.Map = docs => from doc in docs
                           select new
                           {
                               Name = doc.Name,
                               _ = doc.Properties.Select(t => this.CreateField(t.Name, t.Value, new CreateFieldOptions()
                               {
                                   Indexing = FieldIndexing.Exact,
                                   Storage = FieldStorage.No,
                                   TermVector = FieldTermVector.No
                               })),
                               _Default = doc.Properties.Select(t => this.CreateField(t.Name, t.Value, new CreateFieldOptions()
                               {
                                   Indexing = FieldIndexing.Default,
                                   Storage = FieldStorage.No,
                                   TermVector = FieldTermVector.No
                               }))
                           };
    }
}
```
